### PR TITLE
rg-192: Add number of views for resumes

### PR DESCRIPTION
**Pull request by siefimov**
Added number of resume viewes in the Users' resume section on Home page

image

Added number of resume viewes on View Own Resume Page

image

### CODE REVIEW

**Comment by honkster-nazarii** [comment]
add trello link

**Comment by serhiy-yefimov** [comment]
[Number of views for each resume](https://trello.com/c/FV1k3WFa/192-number-of-views-for-each-resume)

**Review by vlad-blip — APPROVED** [review]

**Review by Vitaliy-Kalenichenko — CHANGES_REQUESTED** [review]
The logic for getting the number of views and other data should be implemented on the backend.

**Comment by siefimov** [comment]
I moved the  logic for getting the number of views to the backend.

File: frontend/src/bundles/resume/resume-api.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    public async getViewsCount(
+        userId: string,
+    ): Promise<ResumeViewsCountResponseDto[]> {
+        const response = await this.load(
+            this.getFullEndpoint(ResumesApiPath.USER_ID_VIEWS(userId), {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
```
Comment (Icerock10) [inline frontend/src/bundles/resume/resume-api.ts:51]: If the user `hasAuth`, why then manually pass his `id`?

File: frontend/src/bundles/resume/store/slice.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+type State = {
+    resumeId: string;
+    resumes: ResumeGetItemResponseDto[];
+    resumeViews: ResumeViewsCountResponseDto[];
+};
+
+const initialState: State = {
+    resumeId: '',
+    resumes: [],
```
Comment (Icerock10) [inline frontend/src/bundles/resume/store/slice.ts:17]:
```suggestion
    resumeId: '',
    resumes: [],
```

are you using them somewhere? as I see that on mount only `resumeViews` are fulfilled

**Review by Icerock10 — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/resume/resume-api.ts (hunk 1 above)
Comment (siefimov) [reply frontend/src/bundles/resume/resume-api.ts:49]: I refactored it to use hasAuth

**Review by Valentyn13 — APPROVED** [review]

File: frontend/src/bundles/resume/resume-api.ts [hunk 3]
```diff
@@ -1,45 +1,138 @@
...
+    //         this.getFullEndpoint(`${userIdPath}`, {}),
+    //         {
+    //             method: 'GET',
+    //             contentType: ContentType.JSON,
+    //             hasAuth: true,
+    //         },
+    //     );
+
+    //     return await response.json();
+    // }
```
[the remark this answers is not in the document]
Comment (siefimov) [reply frontend/src/bundles/resume/resume-api.ts]: Removed!

**Comment by Icerock10** [comment]
I'm sorry to forward this task back again, but here is the issue:

image

You are not getting the AWS image url

File: frontend/src/bundles/resume/store/slice.ts [hunk 4]
```diff
@@ -36,6 +39,7 @@ const initialState: State = {
     resumeReview: null,
     templateSettings: null,
     currentTemplateId: null,
+    resumeViews: [],
```
Comment (Icerock10) [inline frontend/src/bundles/resume/store/slice.ts:42]: image

The `resumes` are fulfilled at the same moment as your `resumeViews`, can we just add the views logic to existing resumes? Or if not, you should get AWS image URLs, then fulfilling the resumes is redundant...

**Review by Icerock10 — CHANGES_REQUESTED** [review]

**Comment by siefimov** [comment]
image

I made changes and now:
- the `resumes` are not fulfilled at the same moment with `resumeViews`;
- image has AWS image URLs.

**Review by MaxPoplavskiy — APPROVED** [review]

**Review by Icerock10 — CHANGES_REQUESTED** [review]
image

delete now broken (

conflicts

**Comment by siefimov** [comment]
All works.

**Review by Icerock10 — CHANGES_REQUESTED** [review]
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/assets/24661681/31d5dca6-e54e-46bc-91d5-ea972838579a

the behavior is not correct ( it was removed all at once, I did not investigate much tho :(
resumes are presented and they do not have viewsCount, so on the home page we render resumeViews, but inside editor we are using resumes, kinda mess, resumeViews does not have resumeReview, and currentResume fields which is crucial

**Comment by siefimov** [comment]
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/assets/17623764/db520a79-794d-4f9f-82b5-285a35364dbf

Deleting fixed.
As for resumeViews, please note that `resumes' state collects all resumes from the database. But according to the specification, only resumes of auth user should be displayed on the home page. 
resumeViews flow ensures that only auth user's resumes are retrieved from the database

**Pull request closed without merging**